haskell/private/versions.bzl

```diff
@@ -0,0 +1,60 @@
+# check_version below cannot be called from anywhere
+# because native.bazel_version's availability is restricted:
+#   https://github.com/bazelbuild/bazel/issues/8305
+# An alternative hacky way is as follows:
+#   https://github.com/protocolbuffers/upb/blob/master/bazel/repository_defs.bzl
+#
+# There are utility functions for parsing versions numbers here:
+#   load("@bazel_skylib//lib:versions.bzl", "versions")
+# But we don't want to use them, as skylib is not yet loaded when code
+# in this file executes (and there's no way to execute it later; see first
+# paragraph above).
+
+def _parse_version_chunk(version_chunk):
+    """
+    Args:
+      version_chunk: a chunk of a semantic version, possibly with trailing
+                     content at the end; such as "29", "2", or "3 foobar".
+                     We handle trailing content because
+                     native.bazel_version can: try printing actual_version
+                     in check_version below (got for example
+                     "2.0.0- (@non-git)").
```

Comment:
Nothing to change, just FYI, this is an artifact of the Nix provided Bazel, see https://github.com/NixOS/nixpkgs/issues/80950.